src/Raven.Server/Documents/Indexes/Persistence/Lucene/IndexSearcherHolder.cs

```diff
@@ -155,6 +165,30 @@ public IndexSearcher GetIndexSearcher(IState state)
                 Dispose();
             }
 
+            public void MoveBackToLazy()
+            {
+                if (_lazyIndexSearcher.IsValueCreated == false)
+                    return;
+
+                var old = _lazyIndexSearcher.Value;
+
+                lock (this)
+                {
```

Comment:
Here check that old is the same as the current value, otherwise it was already switched